### PR TITLE
fix: better error handling in account RPCs

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds a check for whether the `account` and `ident` params in several account-based RPCs are strings. If not, it returns `invalidParams`. Some error messages are also improved to be more specific.

Note: This PR looks a lot larger than it is; it is really just the same change(s) applied over and over again across several RPCs.

### Context of Change

rippled currently returns `Internal error` in this scenario. 

The error in the logs is `Caught throw: Type is not convertible to string`

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [x] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

This is a bugfix, not a breaking change, but there isn't an option for that.

## Test Plan
Added unit tests that fail before this patch and now pass.

## Future Tasks
Audit other RPCs for this sort of issue.

### CODE REVIEW

**Comment by mvadari** [comment]
This change could also be done by editing `parseBase58` to return an empty `std::optional` instead of crashing (which would result in an `rpcACT_MALFORMED` instead of `rpcINVALID_PARAMS`, which I believe differs from what Clio returns in this case).

File: src/ripple/rpc/handlers/AccountCurrenciesHandler.cpp [hunk 1]
```diff
@@ -33,19 +33,29 @@ doAccountCurrencies(RPC::JsonContext& context)
 {
     auto& params = context.params;
 
+    if (!(params.isMember(jss::account) || params.isMember(jss::ident)))
+        return RPC::missing_field_error(jss::account);
+
+    std::string strIdent;
```
Comment (godexsoft) [inline src/ripple/rpc/handlers/AccountCurrenciesHandler.cpp:39]: nit: can be avoided via immediately invoked lambda

File: src/ripple/rpc/handlers/AccountInfo.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include <ripple/ledger/ReadView.h>
 #include <ripple/protocol/ErrorCodes.h>
 #include <ripple/protocol/Indexes.h>
+#include <ripple/protocol/RPCErr.h>
```
Comment (godexsoft) [inline src/ripple/rpc/handlers/AccountInfo.cpp:26]: Wasn't all of the `ripple` stuff renamed to `xrpl`? or is it not merged yet? I know there is an alias - i'd refactor the entire codebase to use the new `xrpl` (in a separate PR). surprised it's not already done tbh.

File: src/ripple/rpc/handlers/AccountInfo.cpp [hunk 3]
```diff
@@ -53,9 +54,17 @@ doAccountInfo(RPC::JsonContext& context)
 
     std::string strIdent;
```
Comment (godexsoft) [inline src/ripple/rpc/handlers/AccountInfo.cpp:55]: nit: Same improvement possible

**Review by godexsoft — APPROVED** [review]
Looks fine to me. Leaving a few nits and questions 👍

File: src/ripple/rpc/handlers/AccountCurrenciesHandler.cpp (hunk 1 above)
Comment (mvadari) [reply src/ripple/rpc/handlers/AccountCurrenciesHandler.cpp:39]: Given that the lambda would be handling errors, it'll be more readable etc. if it's left as-is, and probably wouldn't have any performance improvements to switch.

File: src/test/app/AccountTxPaging_test.cpp [hunk 4]
```diff
@@ -271,6 +271,6 @@ class AccountTxPaging_test : public beast::unit_test::suite
     }
 };
 
-BEAST_DEFINE_TESTSUITE(AccountTxPaging, app, ripple);
+BEAST_DEFINE_TESTSUITE(AccountTxPaging, rpc, ripple);
```
Comment (ximinez) [inline src/test/app/AccountTxPaging_test.cpp:274]: Is this change intentional?

File: src/test/app/PayChan_test.cpp [hunk 5]
```diff
@@ -873,6 +873,25 @@ struct PayChan_test : public beast::unit_test::suite
         auto const chan1Str = to_string(channel(alice, bob, env.seq(alice)));
         env(create(alice, bob, channelFunds, settleDelay, pk));
         env.close();
+        {
+            // test account non-string
+            auto testInvalidAccountParam = [&](auto const& param) {
+                Json::Value params;
+                params[jss::account] = param;
+                auto jrr = env.rpc(
+                    "json", "account_tx", to_string(params))[jss::result];
```
Comment (ximinez) [inline src/test/app/PayChan_test.cpp:882]:
```suggestion
                    "json", "account_channels", to_string(params))[jss::result];
```
?

**Review by ximinez — CHANGES_REQUESTED** [review]
Because this branch is based on `master`, `RippledCode.cmake` needs to be updated to reflect the renaming of `AccountLinesRPC_test.cpp` to `AccountLines_test.cpp`.

File: src/test/app/AccountTxPaging_test.cpp (hunk 4 above)
Comment (mvadari) [reply src/test/app/AccountTxPaging_test.cpp:274]: Yes, because it made more sense to me to have it under `rpc` and I figured I'd update it while I was here. If it's a breaking change or anything or just ill advised, I'm happy to change it back.

File: src/test/app/PayChan_test.cpp (hunk 5 above)
Comment (mvadari) [reply src/test/app/PayChan_test.cpp:882]: Typo 🤦

File: src/test/app/AccountTxPaging_test.cpp (hunk 4 above)
Comment (ximinez) [reply src/test/app/AccountTxPaging_test.cpp:274]: Only problem is that we've already got a pending PR (#4794) that updates a bunch of unit tests to make them match their directory names. This would conflict with that, at least in principal, by moving away from that type of naming convention. If you think this makes more sense to be named `rpc`, it might make sense to move the files to `src/test/rpc`.

Comment (mvadari) [reply src/test/app/AccountTxPaging_test.cpp]: Upon further review, I think I confused this test file with the `account_tx` RPC and this test file seems to be testing something else. I rolled back the change.

**Review by ximinez — APPROVED** [review]

**Comment by mvadari** [comment]
Closing in favor of #5078
